### PR TITLE
moving proxy validation function here

### PR DESCRIPTION
**Pull request by djshah19**

### CODE REVIEW

File: pkg/validations/proxy_validation_test.go [hunk 1]
```diff
@@ -52,6 +53,28 @@ func Test_TestProxyUrlConnection(t *testing.T) {
...
+	config.HttpsProxy = proxy.Target.URL
+
+	// valid case
+	err := ValidateProxyParameters(&config)
+	if err != nil {
+		t.Fatalf("Test_ValidateProxyParameters test case failed. %s", err.Error())
+	}
+
+	// invalid case - invalid url
+	config.HttpProxy = "//akse2e:akse2e@skyproxy.ceccloud1.selfhost.corp.microsoft.com:3128"
```
Comment (madhanrm) [inline pkg/validations/proxy_validation_test.go:70]: hope this doesnt cause a credential scanner to raise a bug

Comment (madhanrm) [reply pkg/validations/proxy_validation_test.go:70]: Is this the only proxy server format we are planning to use?

**Review by madhanrm — DISMISSED** [review]

File: pkg/validations/proxy_validation_test.go (hunk 1 above)
Comment (djshah19) [reply pkg/validations/proxy_validation_test.go]: I removed credentials

File: pkg/validations/proxy_validation_test.go [hunk 2]
```diff
@@ -19,7 +20,7 @@ func Test_ValidateProxyURL(t *testing.T) {
 	}
 
 	// Invalid proxy url
-	_, err = ValidateProxyURL("//akse2e:akse2e@skyproxy.ceccloud1.selfhost.corp.microsoft.com:3128")
+	_, err = ValidateProxyURL("//skyproxy.ceccloud1.selfhost.corp.microsoft.com:3128")
```
Comment (raghavendra-nataraj) [inline pkg/validations/proxy_validation_test.go:23]: nit: This repo is public repo. Can we use someother names than our server names

**Review by raghavendra-nataraj — APPROVED** [review]

**Review by msftpgayam — APPROVED** [review]

**Pull request merged**